CorinneTest/Factory.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace CorinneTest
+{
+    class Factory : IPersonFactory
```

Comment:
I'd say that any function in this class that isn't part of the `IPersonFactory` interface signature should be marked `private` or `protected` to keep the scope of the functions within the implementation of this class.  Otherwise, you could imagine another person on your team taking a dependency on the function even though it's not on the interface and that would be bad for maintainability and break the contract.